pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,487 @@
+			parRuntimeQuotaCalculator.UpdateOneGroupRuntimeQuota(parentQuotaInfo)
+		}
+		newSubGroupsTotalRes := parentQuotaInfo.CalculateInfo.Runtime.DeepCopy()
+
+		// 3. update subGroup's cluster resource  when i >= 1 (still has children)
+		if i >= 1 {
+			subTreeWrapper.SetClusterTotalResource(newSubGroupsTotalRes)
+		}
+
+		// 4. update totalRes
+		totalRes = parentQuotaInfo.CalculateInfo.Runtime.DeepCopy()
+	}
+
+	return curToAllParInfos[0].getMaskedRuntimeNoLock()
+}
+
+// updateOneGroupAutoScaleMinQuotaNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) updateOneGroupAutoScaleMinQuotaNoLock(quotaInfo *QuotaInfo, newMinRes v1.ResourceList) {
+	if !quotav1.Equals(quotaInfo.CalculateInfo.AutoScaleMin, newMinRes) {
+		quotaInfo.setAutoScaleMinQuotaNoLock(newMinRes)
+		gqm.runtimeQuotaCalculatorMap[quotaInfo.ParentName].UpdateOneGroupMinQuota(quotaInfo)
+	}
+}
+
+func (gqm *GroupQuotaManager) getCurToAllParentGroupQuotaInfoNoLock(quotaName string) []*QuotaInfo {
+	curToAllParInfos := make([]*QuotaInfo, 0)
+	quotaInfo := gqm.getGroupQuotaInfoByNameNoLock(quotaName)
+	if quotaInfo == nil {
+		return curToAllParInfos
+	}
+
+	for true {
+		curToAllParInfos = append(curToAllParInfos, quotaInfo)
+		if quotaInfo.ParentName == extension.RootQuotaName {
+			break
+		}
+
+		quotaInfo = gqm.getGroupQuotaInfoByNameNoLock(quotaInfo.ParentName)
+		if quotaInfo == nil {
+			return curToAllParInfos
+		}
+	}
+
+	return curToAllParInfos
+}
+
+func (gqm *GroupQuotaManager) GetGroupQuotaInfoByName(quotaName string) *QuotaInfo {
+	gqm.HierarchyUpdateLock.RLock()
+	defer gqm.HierarchyUpdateLock.RUnlock()
+
+	return gqm.getGroupQuotaInfoByNameNoLock(quotaName)
+}
+
+func (gqm *GroupQuotaManager) getGroupQuotaInfoByNameNoLock(quotaName string) *QuotaInfo {
+	return gqm.quotaInfoMap[quotaName]
+}
+
+func (gqm *GroupQuotaManager) getRuntimeQuotaCalculatorByNameNoLock(quotaName string) (treeWrapperPtr *RuntimeQuotaCalculator) {
+	return gqm.runtimeQuotaCalculatorMap[quotaName]
+}
+
+func (gqm *GroupQuotaManager) scopedLockForQuotaInfo(quotaList []*QuotaInfo) func() {
+	listLen := len(quotaList)
+	for i := listLen - 1; i >= 0; i-- {
+		quotaList[i].lock.Lock()
+	}
+
+	return func() {
+		for i := 0; i < listLen; i++ {
+			quotaList[i].lock.Unlock()
+		}
+	}
+}
+
+func (gqm *GroupQuotaManager) UpdateQuota(quota *v1alpha1.ElasticQuota, isDelete bool) error {
+	gqm.HierarchyUpdateLock.Lock()
+	defer gqm.HierarchyUpdateLock.Unlock()
+
+	quotaName := quota.Name
+	updateGroups := make([]*QuotaInfo, 0)
+	if isDelete {
+		_, exist := gqm.quotaInfoMap[quotaName]
+		if !exist {
+			return fmt.Errorf("get quota info failed, quotaName:%v", quotaName)
+		}
+		delete(gqm.quotaInfoMap, quotaName)
+	} else {
+		quotaInfo := NewQuotaInfoFromQuota(quota)
+		// update the local quotaInfo's crd
+		if localQuotaInfo, exist := gqm.quotaInfoMap[quotaName]; exist {
+			localQuotaInfo.UpdateQuotaInfoFromRemote(quotaInfo)
+		}
```

Comment:
if quotaName not in  gqm.quotaInfoMap {
     gqm.quotaInfoMap[name] = NewQuotaInfoFromQuota(quota)
} else {
     quotaInfo := NewQuotaInfoFromQuota(quota)
     gqm.quotaInfoMap[name]. UpdateQuotaInfoFromRemote(quotaInfo)
}